controllers/tab/issueEntry/form/IssueEntryPublicationMetadataForm.inc.php

```diff
@@ -284,11 +289,13 @@ function execute($request) {
 
 			if ($issue) {
 
+				// set publication date
+				$submission->setDatePublished($this->getData('datePublished') ? $this->getData('datePublished') : Core::getCurrentDate());
```

Comment:
The dates are stored for each submission revision (in submission_settings). The date for the latest version is set, wenn this version is published. Unassigning a submission does not change the publication date, because this is at submission level at not at submission revision level. I will add a conceptional description of the versioning and the design decisions we made to the pkp wiki soon, so there will be a place to look stuff like this up...  📝